flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/deployment/AbstractDeploymentReconciler.java

```diff
@@ -39,27 +39,22 @@ public abstract class AbstractDeploymentReconciler implements Reconciler<FlinkDe
 
     private static final Logger LOG = LoggerFactory.getLogger(AbstractDeploymentReconciler.class);
 
-    protected final FlinkOperatorConfiguration operatorConfiguration;
+    protected final FlinkConfigManager configManager;
     protected final KubernetesClient kubernetesClient;
     protected final FlinkService flinkService;
-    protected final Configuration defaultConfig;
 
     public AbstractDeploymentReconciler(
             KubernetesClient kubernetesClient,
             FlinkService flinkService,
-            FlinkOperatorConfiguration operatorConfiguration,
-            Configuration defaultConfig) {
-
+            FlinkConfigManager configManager) {
         this.kubernetesClient = kubernetesClient;
         this.flinkService = flinkService;
-        this.operatorConfiguration = operatorConfiguration;
-        this.defaultConfig = defaultConfig;
+        this.configManager = configManager;
     }
 
     @Override
     public DeleteControl cleanup(FlinkDeployment flinkApp, Context context) {
-        Configuration effectiveConfig = FlinkUtils.getEffectiveConfig(flinkApp, defaultConfig);
-        return shutdownAndDelete(flinkApp, effectiveConfig);
+        return shutdownAndDelete(flinkApp, configManager.getObserveConfig(flinkApp));
```

Comment:
The older code builds config based on current `flinkApp` but IIUC, here `getObserveConfig`  will use `lastReconciledSpec` or `lastStableSpec`. Is this change on purpose?

Reply:
I tried to consistently change it everywhere where we interact with the running job. Because that was technically deployed with either the lastReconciled or lastStable spec as you see it here.